frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/capacity-and-nodes-step/capacity-and-nodes-step.tsx

```diff
@@ -148,46 +105,69 @@ const SelectCapacityAndNodes: React.FC<SelectCapacityAndNodesProps> = ({
 type SelectCapacityAndNodesProps = {
   dispatch: WizardDispatch;
   capacity: WizardState['capacityAndNodes']['capacity'];
-  nodes: WizardState['capacityAndNodes']['nodes'];
+  nodes: WizardState['nodes'];
 };
 
 const SelectedCapacityAndNodes: React.FC<SelectedCapacityAndNodesProps> = ({
   capacity,
   storageClassName,
   enableArbiter,
+  arbiterLocation,
   dispatch,
   nodes,
 }) => {
   const { t } = useTranslation();
   const [pv, pvLoaded, pvLoadError] = useK8sWatchResource<K8sResourceKind[]>(pvResource);
   const memoizedPv = useDeepCompareMemoize(pv);
   const [allNodes, allNodeLoaded, allNodeLoadError] = useK8sWatchResource<NodeKind[]>(nodeResource);
-  const memoizedNodes = useDeepCompareMemoize(allNodes);
+  const memoizedAllNodes = useDeepCompareMemoize(allNodes);
+
+  const pvBySc = React.useMemo(() => getSCAvailablePVs(memoizedPv, storageClassName), [
+    memoizedPv,
+    storageClassName,
+  ]);
 
   React.useEffect(() => {
-    if (pvLoaded && !pvLoadError && memoizedPv.length) {
-      const pvBySc = getSCAvailablePVs(memoizedPv, storageClassName);
+    if (pvLoaded && !pvLoadError) {
       const pvCapacity = calcPVsCapacity(pvBySc);
       const humanizedCapacity = humanizeBinaryBytes(pvCapacity).string;
-      dispatch({ type: 'capacityAndNodes/capacity', payload: humanizedCapacity });
-      dispatch({ type: 'capacityAndNodes/pvCount', payload: memoizedPv.length });
+      dispatch({
+        type: 'capacityAndNodes/capacity',
+        payload: humanizedCapacity,
+      });
+      dispatch({ type: 'capacityAndNodes/pvCount', payload: pvBySc.length });
     }
-  }, [dispatch, memoizedPv, pvLoadError, pvLoaded, storageClassName]);
+  }, [dispatch, pvBySc, pvLoadError, pvLoaded, t]);
 
   React.useEffect(() => {
-    if (memoizedNodes && !allNodeLoadError && memoizedNodes.length && memoizedPv.length) {
-      const pvBySc = getSCAvailablePVs(memoizedPv, storageClassName);
+    if (allNodeLoaded && !allNodeLoadError && memoizedAllNodes.length && pvBySc.length) {
       const pvNodes = getAssociatedNodes(pvBySc);
-      const filteredNodes = memoizedNodes.filter((node) => pvNodes.includes(node.metadata.name));
+      const filteredNodes = memoizedAllNodes.filter((node) => pvNodes.includes(node.metadata.name));
       const nodesData = createWizardNodeState(filteredNodes);
-      dispatch({ type: 'capacityAndNodes/nodes', payload: nodesData });
+      dispatch({ type: 'wizard/nodes', payload: nodesData });
     }
-  }, [dispatch, allNodeLoadError, allNodeLoaded, memoizedNodes, memoizedPv, storageClassName]);
+  }, [dispatch, allNodeLoadError, allNodeLoaded, memoizedAllNodes, pvBySc]);
 
   return (
     <>
+      <StretchCluster
+        enableArbiter={enableArbiter}
+        arbiterLocation={arbiterLocation}
+        onChecked={(isChecked: boolean) =>
+          dispatch({ type: 'capacityAndNodes/enableArbiter', payload: isChecked })
```

Comment:
```suggestion
        onChecked={(isChecked: boolean) =>
          dispatch({ type: 'capacityAndNodes/enableArbiter', payload: isChecked })
```
These anonymous functions will be reevaluated every rerender. Please take them out of here and use `useCallback` with `isChecked` as dependency.

Reply:
Yes, I know. Its nit causing any performance impacts hence I have not added.